server/src/main.rs

```diff
@@ -72,18 +82,23 @@ async fn main() -> Result<(), BoxedStdError> {
     // This simulates the state handling.
     // Once the StartupStateLoader is there, it will be started by the main here and it will send the startup state
     let initial_state_task = tokio::spawn(async move {
-        to_server
-            .update_state(
-                common::commands::CompleteState {
-                    request_id: "".to_owned(),
-                    startup_state: State::default(),
-                    current_state: state,
-                    workload_states: vec![],
-                },
-                vec![],
-            )
-            .await
-            .unwrap_or_illegal_state();
+        if let Some(state) = state {
```

Comment:
We can omit the spawn here by checking beforehand if the state is some.

Reply:
Yes, we can, but this implementation keeps the next line simple.
```
try_join!(communications_task, server_task, initial_state_task).unwrap_or_illegal_state();
```
This way the `try_join` has always three handles. Another approach would be to say that the `initial_state_task` is somehow optional. I am searching for possible solution.

Reply:
Yes, the initial state task is only a workaround for now. We can even omit waiting for it after it is spawned.

Reply:
fixed